riscv_isac/cgf_normalize.py

```diff
@@ -416,34 +416,76 @@ def alternate(var, size, signed=True, fltr_func=None,scale_func=None):
     #return [(coverpoint,"Alternate") for coverpoint in coverpoints]
 
 
-def expand_cgf(cgf_files, xlen):
+def find_duplicates(exp_cp,cgf,coverage,count=0):
+    '''
+    Helper function for checking and printing duplicate coverpoints while inserting into cgf.
+    
+    :param exp_cp: exanded coverpoints
+    :param cgf: the cgf node for current label
+    :param coverage: coverage hit count
+    :param count: current count of duplicate coverpoints
+
+    :type exp_cp: list
+    :type cgf: ordereddict
+    :type coverage: int
+    :type count: int 
+    '''
+    prev_len = len(cgf)
+    for cp,comment in exp_cp:
+        cgf.insert(1,cp,coverage,comment=comment)
+        curr_len = len(cgf)
+        if prev_len == curr_len:
+            count += 1
+            logger.debug("\t#{0}: {1}".format(count,cp))
+        else :
+            prev_len = curr_len
+    return count
+
+def expand_cgf(cgf_files, xlen,list_duplicate=False,cov_labels=None):
     '''
     This function will replace all the abstract functions with their unrolled
     coverpoints
 
     :param cgf_files: list of yaml file paths which together define the coverpoints
     :param xlen: XLEN of the riscv-trace
+    :param list_duplicate: flag to indicate whether or no to log duplicate coverpoints
+    :param cov_labels: If not empty, then expand cgf only for these labels
 
     :type cgf: list
     :type xlen: int
+    :type list_duplicate: bool
+    :type cov_labels: tuple
     '''
+    if cov_labels is not None:
+        cov_labels = list(cov_labels)
+
 
     cgf = utils.load_cgf(cgf_files)
+    if list_duplicate:
+        logger.debug("Found duplicate coverpoints: ")
```

Comment:
Incase there aren't any duplicate coverpoints the logs look misleading if these statements are just printed when the flag is enabled.

Reply:
Should I change it to "the following are the duplicate coverpoints" or shall I remove it altogether?

Reply:
All the relevant information can be printed in the utility function itself (including `function_name` and `covergroup_name`)